sktime/param_est/lag.py

```diff
@@ -0,0 +1,182 @@
+"""Parameter estimators for autoregressive lag order selection.
+
+This module provides functionality for selecting optimal lag orders in
+autoregressive models using information criteria.
+
+The main class ARLagOrderSelector implements lag order selection
+using various information criteria (AIC, BIC, HQIC) and supports both
+sequential and global search strategies.
+"""
+
+__author__ = ["satvshr"]
+__all__ = ["ARLagOrderSelector"]
+
+from sktime.param_est.base import BaseParamFitter
+
+
+class ARLagOrderSelector(BaseParamFitter):
+    """Estimate optimal lag order for autoregressive models using information criteria.
+
+    Implements lag order selection for AR models by comparing
+    different lag specifications using information criteria.
+    Supports both sequential and global search strategies.
+
+    Parameters
+    ----------
+    maxlag : int
+        Maximum number of lags to consider
+    ic : str, default="bic"
+        Information criterion to use for model selection:
+        - "aic" : Akaike Information Criterion
+        - "bic" : Bayesian Information Criterion (default)
+        - "hqic" : Hannan-Quinn Information Criterion
+    glob : bool, default=False
+        If True, searches globally across all lag combinations up to maxlag.
+        If False, searches sequentially by adding one lag at a time.
+    trend : str, default="c"
+        Trend to include in the model:
+        - "n" : No trend
+        - "c" : Constant only
+        - "t" : Time trend only
+        - "ct" : Constant and time trend
+    seasonal : bool, default=False
+        Whether to include seasonal dummies in the model
+    hold_back : int, optional (default=None)
+        Number of initial observations to exclude from the estimation sample
+    period : int, optional (default=None)
+        Period of the data (used only if seasonal=True)
+    missing : str, default="none"
+        How to handle missing values:
+        - "none" : No handling
+        - "drop" : Drop missing observations
+        - "raise" : Raise an error
+
+    Attributes
+    ----------
+    selected_model_ : tuple
```

Comment:
should say `tuple of int`.

Also, mathematically is it really correct that these "minimize the information criterion"? Do they all have equal ic value? Or are these the smallest ones? If so, how many of the smallest ones?

Reply:
So how the function works is that it takes in the max lags and the search strategy you want to use (sequential or global)
and then based on the search strategy it makes pairs of the lags (in the range [1, max lags]) and returns the lag pair with the minimum ic value.